pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -261,7 +261,9 @@ func (jm *ControllerV2) getJobsToBeReconciled(cronJob *batchv1.CronJob) ([]*batc
 
 	for _, job := range jobList {
 		// If it has a ControllerRef, that's all that matters.
-		if controllerRef := metav1.GetControllerOf(job); controllerRef != nil && controllerRef.Name == cronJob.Name {
+		controllerRef := metav1.GetControllerOf(job)
+		if controllerRef != nil && controllerRef.Name == cronJob.Name &&
```

Comment:
Can we rather call the  `resolveControllerRef` function to reuse the logic?

https://github.com/kubernetes/kubernetes/blob/370c85f5ab0b0bfc3b30f235ddb040c246b0e1ff/pkg/controller/cronjob/cronjob_controllerv2.go#L234

Also when looking at resolveControllerRef it could be refactored:
- to receive the cronjob
-  API group check could be added as well in case there was a custom CronJob. However the UID check should catch it even without this check

Reply:
`resolveControllerRef` is to return a `batchv1.cronjob` whereas we have updated `getJobsToBeReconciled` which returns a list of batchv1.job. Though, I have made a change to `getJobsToBeReconciled` to call `resolveControllerRef` and if it returns a cronjob, we add the job in the list of jobs to be returned.